coinExchange/tradeex/controllers/apiusertransmanager.py

```diff
@@ -330,7 +330,16 @@ def on_found_success_purchase_trans(api_trans):
                     notify_resp, comment)
         
         if api_trans.action == API_METHOD_PURCHASE:
+            # [Chi Question] How UserExternalWalletAddress table is used? is below logic works for separate existing tradex traffic with
+            # 3rd party api request?
+            #   My solution is to add an external_crypto_addr column for purchase order created by api user, if it's not DEFAULT_EXTERNAL_CNY_REC_ADDRESS
+            #   then it's created from 3rd party.
             external_crypto_addr = APIUserManager.get_api_user_external_crypto_addr(api_trans.api_user.user.id, 'CNY')
```

Comment:
1) My original idea, is to add the external_cnyf_address in the APIUserTransaction table, not in purchase order.   So that APIUserTransaction holds all the parsed info of the request.
2) This is what I would do,
       a) if APIUserTransaction has the non-empty external_cnyf_address, the external_crypto_addr should be that one
       b) If APIUserTransaction has no external_cnyf_address, the external_crypto_addr should use original logic to be read from ExternalUserWalletAddress table (this was originally allow each personal user to set their external address, we just reuse it for the Investment site api user account, for that scenario this one is used as the default address for the wallet in the Investment site.  But now we want each user in the external api user's app can give us address, so this one will be override now)
       c)  then continue with old logic